File: tests/fixtures/typescript/sample.ts
import { readFileSync } from 'fs';
import path from 'path';
import * as os from 'os';

export interface Shape {
  area(): number;
  perimeter(): number;
}

export interface Describable extends Shape {
  name: string;
  describe(detail: boolean): string;
}

export type Point = {
  x: number;
  y: number;
};

export enum Color { Red, Green, Blue }

export function greet(name: string): string {
  return `Hello, ${name}!`;
}

export function add(a: number, b: number): number {
  return a + b;
}

export class BaseShape {}

export class Circle extends BaseShape implements Shape {
  private label: string = '';

  constructor(private radius: number) {
    super();
  }

  area(): number {
    return Math.PI * this.radius ** 2;
  }

  perimeter(): number {
    return 2 * Math.PI * this.radius;
  }
}

export const multiply = (a: number, b: number): number => a * b;

export const formatPath = (p: string): string => path.normalize(p);

export function convert(value: unknown): string {
  const s = value as string;
  const n = <number>value;
  return s;
}

const items: Point[] = [];
const config: Shape | null = null;

File: tests/fixtures/ruby/sample.rb
require 'json'
require_relative 'shape'

module Geometry
  PI = Math::PI

  class Circle
    attr_reader :radius

    def initialize(radius)
      @radius = radius
    end

    def area
      PI * @radius ** 2
    end

    def perimeter
      2 * PI * @radius
    end

    def self.unit_circle
      new(1.0)
    end
  end
end

def greet(name)
  "Hello, #{name}!"
end

def add(a, b)
  a + b
end

def main
  msg = greet("World")
  total = add(1, 2)
  c = Geometry::Circle.new(5.0)
  puts c.area
end
